src/javascript/app_2/Stores/ui-store.js

```diff
@@ -243,4 +237,16 @@ export default class UIStore extends BaseStore {
     setHasOnlyForwardingContracts(has_only_forward_starting_contracts) {
         this.has_only_forward_starting_contracts = has_only_forward_starting_contracts;
     }
+
+    @action.bound
+    addNotificationBar(message) {
+        if (!this.push_notifications.length) {
+            this.push_notifications.push(message);
+        }
+        this.push_notifications.map(notification => {
```

Comment:
Isn't it going to make duplicates in push_notifications array?

Reply:
true, idk what i was thinking 😅